rfcs/text/testing-refinement.md

```diff
@@ -0,0 +1,143 @@
+
+# Adoption strategy
+
+Developers will need to amend how they write the stories. Examples of various main configurations of the components should be included in one story in the `Design System` directory without any knobs where possible. Existing stories will need to be amended to this format. Other React libraries such as Material-UI (https://material-ui.com) use this method of demonstrating their components. The current standard of documentation in the `Docs` tab will need to be maintained to provide a thorough description of how the component is meant to be used. In addition, these stories will need to be amended where necessary to show events in the `Actions` tab. Developers will need to write one or more additional stories in the `Test` directory for testing purposes presenting as many examples as possible of the components for visual regression purposes. The Cypress tests will need to be compared with the unit tests and, where tests covered by the regression suite can be moved into the unit tests, this should be done. Whilst some duplication will remain where event tests are concerned, we feel it is necessary to have these tests in Cypress too in order to provide a second layer of testing and further confidence that the action triggered by a user is the correct one, so these will remain.
+
+QA Engineers will amend existing specs to leave just the steps to capture snapshots with Applitools for visual verification tests, those testing animations, focus behaviour and events. QA Engineers will need to identify additional scenarios and use cases not testable in the stories in the `Design System` directory or main component list and add them to the stories in the `Test` directory. It would be beneficial if QA Engineers were able to add these themselves with the help of Developers if required.
+
+Applitools can integrate with Jira and automatically log a defect when a test has failed which could be useful and save time with manual intervention, although initial investigation into whether the failure is a valid one should probably be done first so this may not be a feature we could adopt immediately.
+
+# How we teach this
+
+The Testing Strategy `README.md` will be updated with details of this new process. As creating the automation tests now form part of the definition of done for any new components, Developers may need some assistance with understanding how to write the new format of Cypress tests. Support documentation for Applitools is available from https://applitools.com to help with Cypress integration and GitHub. Assistance can also be provided by the Automation Team who have been undertaking the most recent phase of Applitools integration testing with Cypress.
```

Comment:
we should provide correct path to`README.md` file in cypress directory for testing strategy